nomad/server.go

```diff
@@ -410,6 +411,9 @@ func NewServer(config *Config, consulCatalog consul.CatalogAPI) (*Server, error)
 	// Emit metrics
 	go s.heartbeatStats()
 
+	// Emit raft and state store metrics
+	go s.EmitRaftStats(time.Second, s.shutdownCh)
```

Comment:
I see how this matches other stats period, but I suspect that this is too frequent.  Statsd agent commonly only flush data every few seconds seconds (e.g. 10 seconds for datadog [1]), and with gauges, all values in the flush interval are dropped except for the last one [2].

Nothing to change now but raising awareness of potentially wasteful operations here and we can investigate/act later.

[1] https://docs.datadoghq.com/developers/dogstatsd/#how-it-works
[2] https://github.com/statsd/statsd/blob/master/docs/metric_types.md#gauges

Reply:
Would it be more reasonable to use 10seconds here? The one potentially expensive call is `state.LatestIndex` which scans multiple memdb tables to find the max index. I'd rather fix this now to be more conservative about emitting stats.

Reply:
Yeah, I'd go with 10s or 5s, and then we can reexamine other stats periods.